File: 05-BINARY-SEARCH/binary-search.php
<?php

# https://leetcode.com/problems/binary-search/
class Solution
{
    public function search(array $nums, int $target): int
    {
        $low = 0;
        $high = count($nums) - 1;

        while ($low <= $high) {
            $mid = floor(($low + $high) / 2);

            if ($target > $nums[$mid]) {
                $low = $mid + 1;
            } elseif ($target < $nums[$mid]) {
                $high = $mid - 1;
            } else {
                return $mid;
            }
        }
        return -1;
    }
}

File: 05-BINARY-SEARCH/Python/binary-search.py
from typing import *


# https://leetcode.com/problems/binary-search/
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1

        while low <= high:
            mid = (low + high) // 2

            if target > nums[mid]:
                low = mid + 1
            elif target < nums[mid]:
                high = mid - 1
            else:
                return mid

        return -1
